R/estimate_cox.R

```diff
@@ -0,0 +1,195 @@
+#' @title Wrapper for Cox Proportional Hazard analysis
+#'
+#' @description This function is a wrapper around \code{survival::coxph} to fit a Cox Proportional Hazards Regression model, assuming right-censored data.
+#'    The function expects that the data has been filtered on the parameter (PARAM/PARAMCD) of interest.
+#'    The result is an object of class \code{summary.coxph} which can be used in downstream functions and methods that rely on the \code{summary.coxph} class.
+#'
+#' @param data The name of the dataset for the Cox model is based on the Analysis Data Model (ADaM) principles. The dataset is expected to have
+#'    one record per subject per analysis parameter. Rows in which the analysis variable (AVAL) or the sensor variable (CNSR) contain NA, are removed during analysis.
+#' @param main Character vector, representing the covariates used at the Cox model. When NULL, the baseline (null) model is fit.
+#'    Default is NULL.
+#' @param AVAL Analysis value for the Cox model. Default is "AVAL", as per CDISC ADaM guiding principles.
+#' @param CNSR Censor for the Cox model. Default is "CNSR", as per CDISC ADaM guiding principles.
+#' @param conf.int Confidence intervals used for the estimation.
+#' @param ... additional arguments passed on to the ellipsis of the call \code{survival::coxph(data = data, formula = survival::Surv(AVAL, 1-CNSR) ~ main), ...)} .
+#'    Use \code{?survival::coxph} and \code{?survival::summary.coxph} for more information.
+#'
+#' @return a summarized coxph model
+#'
+#' @references \url{https://github.com/therneau/survival}
+#'
+#' @export
+#'
+#' @examples
+#'
+#' ## Null model
+#' visR::estimate_cox(data = adtte)
+#'
+#' ## Univariate Cox regression model using the covariate: `TRTP`
+#' # Stratified Kaplan-Meier analysis by `TRTP`
+#' visR::estimate_cox(data = adtte, main = "TRTP")
+#'
+#' ## Multivariate Cox regression model using the covariates: `TRTP` and `SEX`
+#' visR::estimate_cox(data = adtte, main = c("TRTP", "SEX"))
+#'
+#' ## Stratified Cox model (strata with one level)
+#' visR::estimate_cox(data = adtte, main = c("AGE", "survival::strata(PARAMCD)"))
+#'
+#' ## Null model on subset of adtte
+#' visR::estimate_cox(data = adtte[adtte$SEX == "F", ])
+#'
+#' ## Modify the default analysis by using the ellipsis
+#' visR::estimate_cox(data = adtte, main = c( "SEX", "survival::cluster(USUBJID)"),
+#'   tier = "breslow", conf.int = 0.9)
+#'
+#' ## Example working with non CDISC data
+#' head(survival::veteran)
+#'
+#' # convert time and censoring data to ADaM variables
+#' # convert censoring status to CDISC principles
+#' veteran_adam <- survival::veteran %>%
+#'  dplyr::mutate(AVAL = time,
+#'                CNSR = dplyr::if_else(status == 1, 0, 1)
+#'  )
+#'
+#' visR::estimate_cox(data = veteran_adam, main = c("trt", "diagtime"))
+
+
+
+estimate_cox <- function(
+  data = NULL
+  ,main = NULL
```

Comment:
if you want to put all main, modifiers and interactions into one statment, then opt for consistency with estimate_KM and  call it strata

Reply:
We cannot call it strata because strata we can have stratified parameters which will be denoted with 'strata' in the model. So we cannot have strata for the stratified parameters and strata for the model as well. But other ideas instead of main are more than welcome! :)

Reply:
We can have both. eg model(data=data) Can you give an example of why it would not work?